frontend/projects/upgrade/src/app/features/dashboard/feature-flags/modals/delete-feature-flag-modal/delete-feature-flag-modal.component.ts

```diff
@@ -53,16 +61,28 @@ export class DeleteFeatureFlagModalComponent {
     public dialogRef: MatDialogRef<DeleteFeatureFlagModalComponent>
   ) {}
 
+  ngOnInit(): void {
+    this.listenForFeatureFlagListLengthChanges();
+  }
+
   onInputChange(value: string): void {
     this.inputSubject.next(value);
   }
 
+  // Close the modal once the feature flag list length changes, as that indicates actual success
+  listenForFeatureFlagListLengthChanges(): void {
```

Comment:
maybe update the name to match what you're listening for, listenForFeatureFlagRemoved()